packages/EDA/src/stat-tools.ts

```diff
@@ -0,0 +1,261 @@
+// Statistic tools
+
+/* REFERENCES
+   
+     [1] One-way analysis of variance, https://en.wikipedia.org/wiki/One-way_analysis_of_variance
+
+     [2] G.W. Heiman. Basic Statistics for the Behavioral Sciences, 6th ed. Wadsworth Publishing, 2010
+     
+     [3] F-test of equality of variances, https://en.wikipedia.org/wiki/F-test_of_equality_of_variances
+
+     [4] S. McKillup. Statistics Explained, Cambridge University Press, 2005
+
+*/
+
+import * as grok from 'datagrok-api/grok';
+import * as ui from 'datagrok-api/ui';
+import * as DG from 'datagrok-api/dg';
+
+//@ts-ignore: no types
+import * as jStat from 'jstat';
+
+enum ERROR_MSG {
+  NON_EQUAL_FACTORS_VALUES_SIZE = 'non-equal sizes of factor and values arrays. INPUT ERROR.',
+  INCORRECT_SIGNIFICANCE_LEVEL = 'incorrect significance level. It must be from the interval (0, 1). INPUT ERROR.',
+  INCORRECT_SAMPLE_SIZE = 'incorrect size of sample. DATA FACTORIZAING ERROR.',
+};
+
+export type SampleData = {
+  sum: number,
+  sumOfSquares: number,
+  size: number,
+  data: number[] | undefined | null,
```

Comment:
Strongly-typed raw arrays are always preferred for number crunching. And why | undefined | null?

Reply:
In ANOVA, the array `data ` is not used, just `sum`s are required. So, it can be undefined. Optionally, when splitting values with respect to factor levels (categories), data can be collected. For this, the function `getFactorizedData `should be used instead `getLevelsStat`.
Next, in this particular case, `data` is not specified as a strongly-typed raw array, since its size depends on the `factors` content.

Reply:
> Next, in this particular case, data is not specified as a strongly-typed raw array, since its size depends on the factors content.

Still it's inefficient. Perhaps we need `Column.splitByCategories(Column categorical): Map<String, Float32Array>` method?

Reply:
I think that we do need this method!